core/domain/html_cleaner.py

```diff
@@ -782,3 +791,49 @@ def add_caption_attr_to_image(html_string):
             image['caption-with-value'] = escape_html(json.dumps(''))
 
     return unicode(soup)
+
+
+def add_dimensions_to_noninteractive_image_tag(html_string, exp_id):
+    """Adds dimensions to all oppia-noninteractive-image tags.
+
+    Args:
+        html_string: str. HTML string in which the dimensions is to be
+            added.
+        exp_id: str. Exploration id.
+
+    Returns:
+        str. Updated HTML string with the dimensions for all
+            oppia-noninteractive-image tags.
+    """
+    soup = bs4.BeautifulSoup(html_string.encode('utf-8'), 'html.parser')
+
+    for image in soup.findAll('oppia-noninteractive-image'):
+        filename = unescape_html(image['filepath-with-value'])
+        filename = filename[1:-1]
+        image['filepath-with-value'] = get_filepath_of_object_image(
+            filename, exp_id)
+
+    return unicode(soup)
+
+
+def get_filepath_of_object_image(filename, exp_id):
+    """Gets the dimensions of the image file.
+
+    Args:
+        filename: str. Name of the file whose dimensions need to be
+            calculated.
+        exp_id: str. Exploration id.
+
+    Returns:
+        object. filepath object of the image.
+    """
+    URL = ('https://storage.googleapis.com/%s/%s/assets/image/%s' % (
```

Comment:
This is not really constant.

Reply:
It probably shouldn't be in uppercase.

Reply:
You should probably move the definitions of URLs to *constants,js* since we already use this inside [*AssetsBackendApiService.js*](https://github.com/oppia/oppia/blob/40c53e6cc28b6d1eef612dbd40da335364381cf2/core/templates/dev/head/services/AssetsBackendApiService.js)

Reply:
No need now